File: app/src/main/java/com/likhit/variants/base/BaseFragment.java
package com.likhit.variants.base;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.View;
import android.widget.Toast;

public abstract class BaseFragment extends Fragment implements BaseView {

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
    }

    protected void initViews(View view) {

    }

    protected BaseActivity getBaseActivity() {
        return (BaseActivity) getActivity();
    }

    @Override
    public void showMessage(String message) {
        if (getBaseActivity() != null) {
            Toast.makeText(getBaseActivity(), message, Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void showMessage(int messageResId) {
        if (getBaseActivity() != null) {
            Toast.makeText(getBaseActivity(), getBaseActivity().getString(messageResId), Toast.LENGTH_SHORT).show();
        }
    }
    public boolean onBackPressed() {
        return false;
    }
}

File: app/src/main/java/com/likhit/variants/base/BaseFragment.kt
package com.likhit.variants.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View
import android.widget.Toast

abstract class BaseFragment : Fragment(), BaseView {

    protected val baseActivity: BaseActivity?
        get() = activity as BaseActivity?

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViews(view)
    }

    protected fun initViews(view: View) {

    }

    override fun showMessage(message: String) {
        if (baseActivity != null) {
            Toast.makeText(baseActivity, message, Toast.LENGTH_SHORT).show()
        }
    }

    override fun showMessage(messageResId: Int) {
        if (baseActivity != null) {
            Toast.makeText(baseActivity, baseActivity!!.getString(messageResId), Toast.LENGTH_SHORT).show()
        }
    }

    fun onBackPressed(): Boolean {
        return false
    }
}
